### PR TITLE
98143 Sign-in widget: Update the sign-in alert content 

### PR DESCRIPTION
**Pull request by nichia**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Change the sign-in alert that appears on unauthenticated/Drupal pages needs to reflect that Login.gov and IDme are the two supported login providers, and that identity verification is required

## Related issue(s)

- department-of-veterans-affairs/va.gov-team/issues/98143
- department-of-veterans-affairs/va.gov-team/issues/97279

## Testing done

- Manual testing in local development:
  - 'yarn watch` from `vets-website`
  - `foreman start -m all=1,clamd=0,freshclam=0` from `vets-api`
  - Visit `http://localhost:3001/health-care/schedule-view-va-appointments/` to check that alert content is updated

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |    Screenshot 2024-12-12 at 10 50 15 AM    |   Screenshot 2024-12-12 at 10 52 53 AM    |
| Desktop |   Screenshot 2024-12-12 at 10 51 10 AM     |  Screenshot 2024-12-12 at 10 51 31 AM     |

## What areas of the site does it impact?

These changes impact _widgets_ that get embedded on static pages. The following are some paths that embed widgets that used this alert.

- `/health-care/schedule-view-va-appointments/`
- `/health-care/refill-track-prescriptions/`
- `/health-care/secure-messaging/`
- `/health-care/view-test-and-lab-results/`
- `health-care/get-medical-records/`

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Question: Should I update  ServiceProvidersText.jsx for this PR? I decided not to.
I decided to only update the SignIn alert for `src/applications/static-pages/cta-widget` to avoid breaking the other apps that are presently using `ServiceProvidersText`.

❯ rg ServiceProvidersText -l src 
src/platform/user/authentication/components/ServiceProvidersText.jsx
src/platform/user/package.json
src/platform/user/tests/authentication/components/ServiceProvidersText.unit.spec.js
src/platform/user/exportsFile.js
src/applications/vre/28-1900/containers/ConfirmationPage.jsx
src/applications/financial-status-report/components/shared/UnverifiedPrefillAlert.jsx
src/applications/vre/25-8832/containers/ConfirmationPage.jsx
src/applications/static-pages/download-1095b/components/App/index.js
src/applications/static-pages/medical-copays-cta/components/App/index.js
src/applications/disability-benefits/686c-674-v2/containers/ConfirmationPage.jsx

### CODE REVIEW

**Comment by dcloud** [comment]
> Question: Should I update ServiceProvidersText.jsx for this PR? I decided not to.

I had removed a similar component, `ServiceProvidersList.jsx` and [was asked not to remove that component](https://github.com/department-of-veterans-affairs/vets-website/pull/33298#pullrequestreview-2492694104). I think that rationale from Identity team was that individual components were (somehow) used in Drupal pages (outside of widgets, I guess?). 

Anyway, we should let @department-of-veterans-affairs/octo-identity handle modifying or removing code in `src/platform/user/authentication/components`. So you made the right call!

File: src/applications/static-pages/cta-widget/components/messages/SignIn.jsx [hunk 1]
```diff
@@ -12,31 +9,61 @@ const SignIn = ({
...
+            label={ariaLabel}
+            aria-describedby={ariaDescribedby}
+            uswds
+          />
+        </p>
+        <p>
+          <va-link
+            href="https://www.va.gov/resources/creating-an-account-for-vagov"
+            text="Learn about creating an account"
+            disableAnalytics
```
Comment (radavis) [inline src/applications/static-pages/cta-widget/components/messages/SignIn.jsx:50]: I'm unsure if you want to call `disableAnalytics` here, but the rest of this PR looks good.

**Review by radavis — DISMISSED** [review]

File: src/applications/static-pages/cta-widget/components/messages/SignIn.jsx (hunk 1 above)
Comment (nichia) [reply src/applications/static-pages/cta-widget/components/messages/SignIn.jsx]: Good point. I've removed it.

**Comment by asg5704** [comment]
@nichia I am getting confirmation on this right now in https://dsva.slack.com/archives/C068QM7CCKG/p1733332973315609

If we want the user to sign-in with a verified account we need to use the forced modal

```javascript
// cta-widgets/index.js

// We need to add the forced modal functionality 
openForcedLoginModal = () => this.props.toggleLoginModal(true, '', true);

// Pass it to the SignIn component's primaryButtonHandler
<SignIn primaryButtonHandler={this.openForcedLoginModal} ... the other props />
```
cc: @dcloud

**Comment by dcloud** [comment]
Related PR: department-of-veterans-affairs/vets-website/pull/33521

**Comment by dcloud** [comment]
> @nichia I am getting confirmation on this right now in https://dsva.slack.com/archives/C068QM7CCKG/p1733332973315609
> 
> If we want the user to sign-in with a verified account we need to use the forced modal
> 
> ```js
> // cta-widgets/index.js
> 
> // We need to add the forced modal functionality 
> openForcedLoginModal = () => this.props.toggleLoginModal(true, '', true);
> 
> // Pass it to the SignIn component's primaryButtonHandler
> <SignIn primaryButtonHandler={this.openForcedLoginModal} ... the other props />
> ```
> 
> cc: @dcloud

Per our Slack discussion, I think @asg5704 is saying that some pages that use the CTA widget should not tell veterans that LOA3 authentication is required, and when LOA3 authentication _is_ required, we need to use the `openForcedLoginModal`.

It sounds like a conditional is needed somewhere in the CTA widget, and might be in place somewhere already

**Review by asg5704 — APPROVED** [review]

**Review by dcloud — APPROVED** [review]

**Comment by fmccaf1** [comment]
Looks good @nichia !

**Review by IGallupSoCo — APPROVED** [review]
LGTM

**Pull request merged**